core/compress_api.php

```diff
@@ -83,6 +83,13 @@ function compress_handler_is_enabled() {
  * @access public
  */
 function compress_start_handler() {
+	# Do not start compress handler if we got any output so far, as this
+	# denotes that an error has occurred. Enabling compression in this case
+	# will likely cause a Content Encoding Error when displaying the page.
+	if( ob_get_length() ) {
```

Comment:
I don't think performance is an issue given that ob_get_length() will typically be 0.  If I'm missing something and there is a performance issue, would @ob_start() fix the error surfacing?  Would setting a flag in trigger_error() and checking it here help?